ros/src/twist_controller/dbw_node.py

```diff
@@ -101,61 +95,60 @@ def __init__(self):
                            'max_lat_acc': max_lat_accel,
                            'max_steer_angle': max_steer_angle
                           }
+
         self.controller = Controller(**controller_args)
         self.yaw_controller = YawController(
             wheel_base, steer_ratio, min_speed, max_lat_accel, max_steer_angle)
 
         # Subscribe to all the topics you need to
-        rospy.Subscriber('/vehicle/dbw_enabled', Bool,
-                         self.dbw_cb, queue_size=1)
-        rospy.Subscriber('/final_waypoints', Lane,
-                         self.waypoints_cb, queue_size=1)
-        rospy.Subscriber('/current_pose', PoseStamped,
-                         self.pose_cb, queue_size=1)
-        rospy.Subscriber('/current_velocity', TwistStamped,
-                         self.velocity_cb, queue_size=1)
-        rospy.Subscriber('/twist_cmd', TwistStamped,
-                         self.twist_cb, queue_size=1)
+        rospy.Subscriber('/vehicle/dbw_enabled', Bool, self.dbw_cb, queue_size=1)
+        rospy.Subscriber('/final_waypoints', Lane, self.waypoints_cb, queue_size=1)
+        rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb, queue_size=1)
+        rospy.Subscriber('/current_velocity', TwistStamped, self.velocity_cb, queue_size=1)
+        rospy.Subscriber('/twist_cmd', TwistStamped, self.twist_cb, queue_size=1)
 
         self.loop()
 
     def loop(self):
-        """
-        The main functionallity of the DBW Node.
-        It runs at 50Hz = 50 cycles per second, until ROS shuts down.
-        It reads the required velocities and other flags, creates and publishes
-        if required the contoller commands
-        """
+        """The main functionallity of the DBW Node."""
         rate = rospy.Rate(10)  # 10Hz
+
         while not rospy.is_shutdown():
+
             data = [self.velocity, self.waypoints, self.pose]
             all_available = all([x is not None for x in data])
 
             if not all_available:
                 continue
-            # check if too few waypoints and publish a hard break
-            if len(self.waypoints) < dbw_helper.POINTS_TO_FIT:
-                rospy.logwarn("Number of waypoint received: %s",
-                              len(self.waypoints))
-                throttle, brake, steer = 0, -5, 0
-            else:
+
+            if len(self.waypoints) >= dbw_helper.POINTS_TO_FIT:
+
                 # Read target and current velocities
                 cte = dbw_helper.cte(self.pose, self.waypoints)
                 target_velocity = self.waypoints[0].twist.twist.linear.x
                 current_velocity = self.velocity.linear.x
                 vel_error = target_velocity - current_velocity
-                reference_velocity = 15 #mph
+                reference_velocity = 10 #mph
```

Comment:
Can you use here a Constant as you did in the waypoint_updater.py
In case it changes later, not to forget it.

Reply:
Sure @mauricelos made that change :)